docs/kubernetes/deploy-to-google-cloud-from-ci.md

```diff
@@ -0,0 +1,113 @@
+# Deploy your application to Google Cloud on your CI/CD
+This tutorial describes how to setup your CI/CD to deploy your application to Google Cloud.
+
+For this example we will be using Jenkins as our CI.
+
+## Kubernetes Buildpack
+To deploy project to Google Cloud, you need couple of technologies, to not bother users with installing these technologies, we created a [shopsys/kubernetes-buildpack](https://github.com/shopsys/kubernetes-buildpack) image, which can be used for deployment.
+
+## Scripts
+### Deploy to Kubernetes
+Script used for building current state of and application into google cloud
+
+[deploy-to-kubernetes.sh](/project-base/.ci/deploy-to-kubernetes.sh)
+
+#### Environment variables
+| Variable name               | Description
+| -------------               |-------------
+| DOCKER_USERNAME             | docker login
+| DOCKER_PASSWORD             | docker password
+| FIRST_DOMAIN_HOSTNAME       | domain hostname for first domain - without http://
+| SECOND_DOMAIN_HOSTNAME      | domain hostname for second domain - without http://
+| FIRST_DOMAIN_SSL_DIRECTORY  | path to SSL certificates for first domain
+| SECOND_DOMAIN_SSL_DIRECTORY | path to SSL certificates for second domain
+| PROJECT_ID                  | project id of your google project
+| GIT_COMMIT                  | commit hash to build your images uniquely
+| WORKSPACE                   | project root dir
+
+#### Mounts
+
+##### Docker Socket
+Docker socket is used to build and push image of php-fpm
+```
+-v /var/run/docker.sock:/var/run/docker.sock \
+```
+
+##### Terraform state
+Terraform state mounted locally to be able to apply changes or destroy terraform infrastructure.
+
+```
+-v ~/google-cloud/.terraform/tfstate:/tmp/infrastructure/google-cloud/tfstate \
+```
+
+##### SSL Certificates
+SSL certificates for secured protocol, mount these into folder of your choice that you choose setting environment variable.
+
+```
+-v ~/path/to/certificates-1:$FIRST_DOMAIN_SSL_DIRECTORY \
+-v ~/path/to/certificates-2:$SECOND_DOMAIN_SSL_DIRECTORY \
+```
+
+##### Google Account Service
+Mount your service-account.json obtained from google-cloud
+
+```
+-v ~/google-cloud/service-account.json:/tmp/infrastructure/google-cloud/service-account.json \
+```
+#### Usage
+Use Kubernetes buildpack with environment variable set:
```

Comment:
I would add a sentence that explains what is the final state, e.g. "After running the following script, you will have your application running in Google Cloud" or something like that